e2e/kube.go

```diff
@@ -1 +1,151 @@
 package e2e
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/agentbakere2e/config"
+	"k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+type Kubeclient struct {
+	Dynamic client.Client
+	Typed   kubernetes.Interface
+	Rest    *rest.Config
+}
+
+func newKubeclient(config *rest.Config) (*Kubeclient, error) {
+	dynamic, err := client.New(config, client.Options{})
+	if err != nil {
+		return nil, fmt.Errorf("create dynamic Kubeclient: %w", err)
+	}
+
+	restClient, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, fmt.Errorf("create rest kube client: %w", err)
+	}
+
+	typed := kubernetes.New(restClient)
+
+	return &Kubeclient{
+		Dynamic: dynamic,
+		Typed:   typed,
+		Rest:    config,
+	}, nil
+}
+
+func getClusterKubeClient(ctx context.Context, resourceGroupName, clusterName string) (*Kubeclient, error) {
+	data, err := getClusterKubeconfigBytes(ctx, resourceGroupName, clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("get cluster kubeconfig bytes: %w", err)
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(data)
+	if err != nil {
+		return nil, fmt.Errorf("convert kubeconfig bytes to rest config: %w", err)
+	}
+	restConfig.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: scheme.Codecs}
```

Comment:
why do we need this extra configuration?

Reply:
I just moved the code. Don't know the original intent

Reply:
ah so this was already here...